arangod/RocksDBEngine/CMakeLists.txt

```diff
@@ -1,107 +1,126 @@
+  RocksDBVPackIndex.cpp
+  RocksDBValue.cpp
+  RocksDBWalAccess.cpp
+  RocksDBZkdIndex.cpp
+  ReplicatedRocksDBTransactionCollection.cpp
+  ReplicatedRocksDBTransactionState.cpp
+  SimpleRocksDBTransactionState.cpp
+  SortedRowsStorageBackendRocksDB.cpp)
+
+target_link_libraries(arango_rocksdb
+  arango_cache
+  arango_common_rest_handler
+  arango_geo
+  arango_health
+  arango_indexes
+  arango_restart_action
+  arango_storage_engine_common
+  arango::validation
+  arango_zkd
+  boost_boost
+  rocksdb
+  rocksdb_interface
+  snappy)
+
+set(SNAPPY_INCLUDE_DIR "${SNAPPY_SOURCE_DIR};${SNAPPY_BUILD_DIR}" CACHE PATH "relation to snappy")
+target_include_directories(arango_rocksdb PUBLIC ${SNAPPY_INCLUDE_DIR})
+
+if (USE_ENTERPRISE)
+  # this is required for hotbackup. Views need to be flushed.
+  target_include_directories(arango_rocksdb PUBLIC
+    ${IRESEARCH_INCLUDE})
+	target_link_libraries(arango_rocksdb
+    arango_rclone)
+endif()
+
+if ((CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG) AND NOT BUILDING_FOR_ARM64)
+  target_compile_options(arango_rocksdb PUBLIC "-maes")
+endif ()
 
-##
 ## these defines are copied from rocksdb's CMakeLists.txt
 ##  because we are currently using some of their internal headers
 ##  in our build
-# add definitions for various operating systems
 if(CMAKE_SYSTEM_NAME MATCHES "Cygwin")
-  add_definitions(-fno-builtin-memcmp -DCYGWIN)
+  target_compile_options(arango_rocksdb PRIVATE -fno-builtin-memcmp -DCYGWIN)
 elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
-  add_definitions(-DOS_MACOSX)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_MACOSX)
 elseif(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  add_definitions(-DOS_LINUX)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_LINUX)
 elseif(CMAKE_SYSTEM_NAME MATCHES "SunOS")
-  add_definitions(-DOS_SOLARIS)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_SOLARIS)
 elseif(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-  add_definitions(-DOS_FREEBSD)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_FREEBSD)
 elseif(CMAKE_SYSTEM_NAME MATCHES "NetBSD")
-  add_definitions(-DOS_NETBSD)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_NETBSD)
 elseif(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
-  add_definitions(-DOS_OPENBSD)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_OPENBSD)
 elseif(CMAKE_SYSTEM_NAME MATCHES "DragonFly")
-  add_definitions(-DOS_DRAGONFLYBSD)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_DRAGONFLYBSD)
 elseif(CMAKE_SYSTEM_NAME MATCHES "Android")
-  add_definitions(-DOS_ANDROID)
+  target_compile_options(arango_rocksdb PRIVATE -DOS_ANDROID)
 elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
-  add_definitions(-DWIN32 -DOS_WIN -D_MBCS -DWIN64 -DNOMINMAX)
+  target_compile_options(arango_rocksdb PRIVATE -DWIN32 -DOS_WIN -D_MBCS -DWIN64 -DNOMINMAX)
   if(MINGW)
-    add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_VISTA)
+    target_compile_options(arango_rocksdb PRIVATE -D_WIN32_WINNT=_WIN32_WINNT_VISTA)
   endif()
 endif()
```

Comment:
I'm not sure if these definitions are really needed: I restricted them to only the arango_rocksdb library, before they were added to all libraries in arangod.